mapreader/download/sheet_downloader.py

```diff
@@ -0,0 +1,1097 @@
+    ) -> None:
+        """
+        Iniitalise SheetDownloader class
+
+        Parameters
+        ----------
+        metadata_path : str
+            path to metadata.json
+        download_url : Union[str, list]
+            The base URL pattern used to download tiles from the server. This
+            should contain placeholders for the x coordinate (``x``), the y
+            coordinate (``y``) and the zoom level (``z``).
+        """
+        self.polygons = False
+        self.grid_bbs = False
+        self.wfs_id_nos = False
+        self.published_dates = False
+        self.found_queries = []
+        self.merged_polygon = None
+
+        assert isinstance(
+            metadata_path, str
+        ), "[ERROR] Please pass metadata_path as string."
+
+        if os.path.isfile(metadata_path):
+            with open(metadata_path, "r") as f:
+                self.metadata = json.load(f)
+                self.features = self.metadata["features"]
+                print(self.__str__())
+
+        else:
+            raise ValueError("[ERROR] Metadata file not found.")
+
+        if isinstance(download_url, str):
+            my_ts = [download_url]
+        elif isinstance(download_url, list):
+            my_ts = download_url
+        else:
+            raise ValueError(
+                "[ERROR] Please pass ``download_url`` as string or list of strings."
+            )
+
+        self.tile_server = my_ts
+        
+        crs_string = CRS(self.metadata["crs"]["properties"]["name"])
+        self.crs = crs_string.to_string()
+        
+    def __str__(self) -> str:
+        info = f"[INFO] Metadata file has {self.__len__()} item(s)."
+        return info
+
+    def __len__(self) -> int:
+        return len(self.features)
+
+    def get_polygons(self) -> None:
+        """
+        For each map in metadata, creates a polygon from map geometry and saves to ``features`` dictionary.
+        """
+        for feature in self.features:
+            polygon = shape(feature["geometry"])
+            map_name = feature["properties"]["IMAGE"]
+            if len(polygon.geoms) != 1:
+                f"[WARNING] Multiple geometries found in map {map_name}. Using first instance."
+            feature["polygon"] = polygon.geoms[0]
+
+        self.polygons = True
+
+    def get_grid_bb(self, zoom_level: Optional[int] = 14) -> None:
+        """
+        For each map in metadata, creates a grid bounding box from map polygons and saves to ``features`` dictionary.
+
+        Parameters
+        ----------
+        zoom_level : int, optional
+            The zoom level to use when creating the grid bounding box.
+            Later used when downloading maps, by default 14.
+        """
+        if not self.polygons:
+            self.get_polygons()
+
+        if self.crs != "EPSG:4326":
+            raise NotImplementedError("[ERROR] At the moment, MapReader can only create grid bounding boxes and download map sheets projected in EPSG 4326.")
```

Comment:
I'm very happy that this is an explicit error message. However, for accuracy/pedantry, the error message should be:

"[ERROR] At the moment, MapReader can only create grid bounding boxes and query map sheets, using coordinates in WGS1984 (aka EPSG:4326)."

Reply:
Have done this but also made an extra ticket for us to (at some point) make it possible to use other CRSs by converting to EPSG:4326 in the code.